deeplearning4j/deeplearning4j-ui-parent/deeplearning4j-vertx/src/main/java/org/deeplearning4j/ui/VertxUIServer.java

```diff
@@ -302,11 +401,28 @@ private void modulesViaServiceLoader(List<UIModule> uiModules) {
     }
 
     @Override
-    public void stop() {
-        server.close();
-        shutdown.set(true);
+    public void stop() throws InterruptedException {
+        CountDownLatch l = new CountDownLatch(1);
+        Promise<Void> promise = Promise.promise();
+        promise.future().compose(
+                successEvent -> Future.future(prom -> l.countDown()),
+                failureEvent -> Future.future(prom -> l.countDown())
+        );
+        /**
+         * Stop Vertx instance and release any resources held by it.
+         * Pass promise to {@link #stop(Promise)}.
+         */
+        vertx.close(ar -> promise.handle(ar));
+        // synchronous function should wait until the server is stopped
+        l.await();
     }
 
+    @Override
+    public void stop(Promise<Void> stopCallback) {
+        shutdown.set(true);
+        stopCallback.complete();
+        log.info("Deeplearning4j UI server stopped.");
+    }
```

Comment:
This method looks off to me.
If I (as a user) call `stop()` - everything is fine. If I call `stop(Promise<Void>)` - the vertx instance isn't actually stopped?

Reply:
Well, I did not test this scenario, i.e. calling directly `UIServer.stop(Promise<Void>)`. This method is called by `vertx.close()`.
The documentation of `Verticle.stop(Promise<Void>)` tells that:
> Vert.x calls this method when un-deploying the instance. You do not call it yourself.

Adding a separate `UIServer.stopAsync(Promise<Void>)` function and making `VertxUIServer.stopAsync(Promise<Void>)` private would clarify this.

Reply:
I fixed async stopping of UIServer.